components/atom/switch/src/index.scss

```diff
@@ -1,127 +1,142 @@
-    transition: background-color, border 0.4s ease-in-out;
-    width: $w-atom-switch--m;
+    height: $h-atom-switch-container-default;
+    transition: all 0.4s ease-in-out;
+    width: $w-atom-switch-container-default;
 
-    .sui-AtomSwitch--circle {
+    .sui-AtomSwitch-circle {
       background-color: $c-white;
-      border-radius: ($h-atom-switch-container--m - 2) / 2;
+      border-radius: ($h-atom-switch-container-default - 2) / 2;
       border: solid 1px $c-white;
       box-shadow: 2px 0 2px 0 $c-gray-light;
-      height: $h-atom-switch-container--m - 2;
+      height: $h-atom-switch-container-default - 2;
       transition: margin 0.4s ease-in-out;
-      width: $h-atom-switch-container--m - 2;
+      width: $h-atom-switch-container-default - 2;
 
       &.sui-AtomSwitch--toggle {
-        margin-left: $m-atom-switch-circle--m;
+        box-shadow: -2px 0 2px 0 $c-atom-switch--active-shadow;
+        margin-left: $m-atom-switch-circle-default--active;
       }
     }
   }
 
   &.sui-AtomSwitch--disabled {
     color: $c-gray;
     pointer-events: none;
+    .sui-AtomSwitch-inputContainer {
+      border: solid 1px #f1f1f1;
+      background-color: #f1f1f1;
+    }
+    .sui-AtomSwitch-circle {
+      background-color: #f1f1f1;
+      background-color: #f1f1f1;
+    }
   }
 
   &.sui-AtomSwitch--active {
-    & .sui-AtomSwitch--container {
-      .sui-AtomSwitch--inputContainer {
+    & .sui-AtomSwitch-container {
+      .sui-AtomSwitch-inputContainer {
         background-color: $c-atom-switch--active;
-        border-radius: 11px;
         border: solid 1px $c-atom-switch--active;
         cursor: pointer;
       }
     }
+    & .sui-AtomSwitch-circle {
+      box-shadow: 2px 0 2px 0 $c-atom-switch--active-shadow;
+    }
   }
 
-  &.sui-AtomSwitch--large, &.sui-AtomSwitch--active.sui-AtomSwitch--large {
-    .sui-AtomSwitch--container {
-      height: $h-atom-switch--l;
+  &.sui-AtomSwitch-large, &.sui-AtomSwitch--active.sui-AtomSwitch-large {
+    .sui-AtomSwitch-container {
+      height: $h-atom-switch-container-large;
       padding-bottom: $p-s;
       padding-top: $p-s;
-      .sui-AtomSwitch--text {
+      .sui-AtomSwitch-text {
         display: flex;
         flex-direction: column;
         justify-content: center;
       }
 
-      .sui-AtomSwitch--inputContainer {
-        border-radius: $w-atom-switch--l / 2;
-        height: $h-atom-switch-container--l;
-        width: $w-atom-switch--l;
-        .sui-AtomSwitch--circle {
-          border-radius: ($h-atom-switch-container--l - 2) / 2;
-          height: $h-atom-switch-container--l - 2;
-          width: $h-atom-switch-container--l - 2;
+      .sui-AtomSwitch-inputContainer {
+        height: $h-atom-switch-inputContainer-large;
+        width: $w-atom-switch-container-large;
+        .sui-AtomSwitch-circle {
+          border-radius: $w-atom-switch-container-large / 2;
+          height: $h-atom-switch-inputContainer-large - 2;
```

Comment:
To me, the repeated `-2` is border-width * 2, so I would use a variable for that. Just an opinion.